magma/loong.patch

```diff
@@ -0,0 +1,89 @@
+diff --git a/PKGBUILD b/PKGBUILD
+index 2ef55de..b9fc950 100644
+--- a/PKGBUILD
++++ b/PKGBUILD
+@@ -7,7 +7,7 @@
+ 
+ _pkgname=magma
+ pkgbase=$_pkgname
+-pkgname=(magma-cuda magma-hip)
++pkgname=(magma-hip)
+ pkgver=2.9.0
+ pkgrel=1
+ _pkgdesc="Matrix Algebra on GPU and Multicore Architectures"
+@@ -16,7 +16,6 @@ url="https://icl.utk.edu/magma/"
+ license=('BSD-3-Clause')
+ depends=('blas' 'lapack')
+ makedepends=('git' 'cmake' 'ninja' 'python' 'gcc-fortran'
+-             'cuda'
+              'rocm-core' 'hip-runtime-amd' 'hipblas' 'hipsparse')
+ optdepends=('python: for examples and tests'
+             'gcc-fortran: Fortran interface')
+@@ -42,33 +41,14 @@ prepare() {
+ 
+   cd "${srcdir}"
+ 
+-  cp -r "${_pkgname}" "${_pkgname}-${pkgver}-cuda"
+   cp -r "${_pkgname}" "${_pkgname}-${pkgver}-hip"
+ 
+-  cd "${_pkgname}-${pkgver}-cuda"
+-  echo -e "BACKEND = cuda\nFORT = true\nGPU_TARGET=$(_valid_sm)" > make.inc
+-  make generate
+-
+-  cd "${srcdir}"
+-
+   cd "${_pkgname}-${pkgver}-hip"
+   echo -e "BACKEND = hip\nFORT = true\nGPU_TARGET=$(_valid_gfx)" > make.inc
+   make generate
+ }
+ 
+ build() {
+-  echo "Build with cuda backend"
```

Comment:
There are a lot of deleted content here. In order to avoid patch conflicts, it is recommended to follow the method of https://wszqkzqk.github.io/2024/08/22/loongarchlinux-port-tips/#%E5%AF%B9pkgbuild%E5%A4%A7%E6%AE%B5%E5%88%A0%E9%99%A4%E7%9A%84%E5%A4%84%E7%90%86